src/main/java/jenkins/bouncycastle/api/SecurityProviderInitializer.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package jenkins.bouncycastle.api;
+
+import java.security.Security;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+
+import hudson.Plugin;
+import hudson.init.InitMilestone;
+import hudson.init.Initializer;
+
+/**
+ * Initialization class to register Bouncy Castle as a security provider.
+ *
+ */
+public class SecurityProviderInitializer extends Plugin {
```

Comment:
We also need a `MasterToSlaveCallable` that will register BouncyCastle on the slave... perhaps that can be hidden and just expose a utility method that takes a `Channel` and will return when BouncyCastle is registered on the remote side... that way it can save a `ChannelProperty` on the local side of the channel to flag that BC has been registered so that a round-trip can be avoided for repeated calls and a lock can be managed in case multiple plugins try to register in parallel